File: testdata/jennies/builders/foreign_builder/JavaBuilder/builder_pkg/BuilderPkgSomeNiceBuilderBuilder.java
package builder_pkg;

import some_pkg.SomeStruct;

public class BuilderPkgSomeNiceBuilderBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public BuilderPkgSomeNiceBuilderBuilder() {
        this.internal = new SomeStruct();
    }
    public BuilderPkgSomeNiceBuilderBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/foreign_builder/CSharpBuilder/src/Grafana/Foundation/BuilderPkg/BuilderPkgSomeNiceBuilderBuilder.cs
namespace Grafana.Foundation.BuilderPkg;

using Grafana.Foundation.SomePkg;

public class BuilderPkgSomeNiceBuilderBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public BuilderPkgSomeNiceBuilderBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public BuilderPkgSomeNiceBuilderBuilder Title(string title)
    {
        this.@internal.Title = title;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
